fix 安全加固英文文档，并且去除海外安全加固文档 (#863)
taptap/tds-docs: i18n/en/docusaurus-plugin-content-docs/current/sdk/taptap-appsafety/guide.mdx

---
title: Application Security Access Guide
sidebar_label: Guide
sidebar_position: 2
---

This article describes how to add the app security feature introduced by TapTap to your game.

## Description of APK hardening function

### Functional features

| Function Name | Descriptions | 
| --- | --- |

> | SO Encryption Protection | Encrypt and obfuscate the code, function and export table of the SO file to prevent HOOK attacks and effectively improve the difficulty of shelling. |

Reviewer: ```suggestion
| SO Encryption Protection | Encrypt and obfuscate the code, functions and export tables of the SO files to prevent HOOK attacks and effectively improve the difficulty of unpacking. |
```

| global-metadata Encrypted |For IL2CPP games, the global-metadata is hardened to raise the cracking threshold. |
| AAB Reinforcement | Support for Android app bundle hardening solutions. |

> | Reinforcement programme self-protection | Obfuscating the source code of the augmentation scheme greatly raises the threshold of the plug-in author's reverse analysis of the augmentation scheme, further increasing the difficulty of cracking the game. |

Reviewer: ```suggestion
| Reinforcement programme self-protection | The source code of the reinforcement scheme is obfuscated, which greatly raises the threshold for reverse analysis of the reinforcement scheme by plug-in authors and further increases the difficulty of cracking the game. |
```

| ROOT Environmental Testing | Reduce the potential for fraud in high-risk environments. |
| Debugging Against | Detection of debugging tools such as IDA, FRIDA, etc., effectively raising the threshold for dynamic analysis. |
| Simulator Recognition | Based on a sample library, it accurately identifies simulator users and provides an accurate basis for game segregation matching. |
| Cloud Mobile Phone Recognition | Multi-dimensional determination of the cloud phone environment based on real phone data and multiple cloud phone samples from the extranet. |
| Anti Secondary Packaging | To keep the signature consistent before and after reinforcement, you can enable signature verification to prevent secondary packaging. |
| Malware Countermeasure | Anti-frame plug-ins, memory modifiers, multi-openers and other software. |

### Operating Instructions

#### Function Entrance

> - For games that have not joined the Payroll programme and have exclusive hairstyles on TapTap, you can access them directly through the **Application Security** portal, or contact the relevant business or operations department to apply for the programme if there is no such portal.

Reviewer: ```suggestion
- Games that have not joined the Fire Plan and have exclusive published on TapTap can be entered directly through the **Application Security** portal, or contact the relevant business or operation departments to apply for it if there is no such portal.
```

> - For games on the Payroll programme, you can select the APK enhancement rights directly from the Payroll page and click Use Now (as shown below).

Reviewer: ```suggestion
- For games joined the Fire Plan, you can select the APK enhancement rights directly from the Fire Plan page and click Use Now (as shown below).
```

![](https://capacity-files.lcfile.com/wKzxfJfVfKOxDKRlEgiQvAnQ69OdKiak/app-safety-jiagu.png)

#### Rules

> - The number of successful reinforcement attempts per game per natural month is limited to 5, so developers should not abuse the service;

Reviewer: ```suggestion
- The number of successful reinforcement attempts for each game in a single natural month is not allowed to exceed 5 times. Developers should not abuse the service;
```

- For users with multi-channel requirements, please ensure that the project is in normal operation if reinforcement is required before secondary packaging. The size of the backup file should not exceed 2G;
- The size of the augmentation file should not exceed 2G;
- Reinforcement does not support automatic signing and you must re-sign after reinforcement;
- You can only reinforce a maximum of 2 files each time.

## Description of the anti-cheating function

> The anti-cheat detection function is mainly used to display the data of illegal operation and the environment of the platform that monitors and captures the application when it is running after the game has passed the reinforcement. By displaying cheating devices and related details, developers can gain a deep understanding of the current security situation and accurately grasp the security posture.

Reviewer: ```suggestion
The anti-cheating detection function mainly means that after the game is reinforced, the platform will monitor and capture data on illegal operations and environment when the app is running for display. By displaying cheating devices and related details, developers can gain a deep understanding of the current security situation and accurately grasp the security situation.
```

![](https://capacity-files.lcfile.com/XJgCXueaiMHdcKDHjom7yBel6fqL8zD7/app-safety-resist.png)

### Page Field Descriptions

- UUID: running device tag
- UserID: reported on the game page
- Feature number: hexadecimal display value

<table>
 	<tr>
		<th> Feature Code </th>
		<th> ID </th>
		<th>  Detection Type </th>
		<th> Detailed Information </th>
	</tr>
	<tr>
		<td rowspan="2"> 1000000000000000+id </td>
		<td> 0 </td>
		<td> root class </td>

> 		<td> root root system, test signature pattern </td>

Reviewer: ```suggestion
		<td> root system, test signature pattern </td>
```

		<td> 0 </td>
		<td> Generic Plug-ins, Peripherals Classes </td>
		<td>  </td>
	</tr>
	<tr>
		<td> any </td>
		<td>  Generic Plug-ins, Peripherals type value</td>
		<td> Peripheral </td>
	</tr>

  <tr>
		<td rowspan="2"> B000000000000000+id </td>
		<td> 0 </td>
		<td> Sandbox Classes </td>
		<td>  </td>
	</tr>
	<tr>
		<td> any </td>
		<td> Sandbox Detection Type </td>
		<td> Type </td>
	</tr>
</table>

- Detection type

| index | type | description | 
| --- | --- | --- |
| 1 | Root Class | android and ios: running environment is root environment, windows: test signature environment |
| 2 | Debugging Class | Running processes are debugged |
| 3 | Hook Class |  hooked |
| 4 | Memory Modifier Class | Running content has been modified |
| 5 | Resource Modification Class | Resource signatures have been modified |
| 6 | Simulator Class | Simulator running |
| 7 | Cloud Provider Class | Cloud real machine operation |
| 8 | General Plug-ins | Known types of cheating software detection |
| 9 | Special Plug-ins | Peripheral Input Capture Calculation |
| 10 | Injection Class | 	Cheat frames or other injections |
| 11 | Sandbox Class | Sandbox environment operation |

- Detection results

> 1. Implement processing strategies according to the detection type.

Reviewer: ```suggestion
1. Implement processing strategies according to detection types.
```

2、Detect to stop or do not perform additional operations.

- Detailed information

1、Detection type details
 

### Function usage rules

#### Diagram operation

- Provide filter function based on report ID, user ID, detection type, detection result and time.
- Provide report download function to support downloading the detected data to the local area.

![](https://capacity-files.lcfile.com/skGGCntpLlMUqjeQOMESC2mHQs7dOG9q/app-safety-diagrams.png)

### Anti-cheat strategy configuration

> - Provide anti-cheat strategy configuration, developers can configure their own anti-cheat strategy.

Reviewer: ```suggestion
- Anti-cheating strategy configuration is provided, and developers can configure their own anti-cheating strategies.
```
